bundles/ui/org.eclipse.smarthome.ui.icon/src/main/java/org/eclipse/smarthome/ui/icon/IconProvider.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.ui.icon;
+
+import java.io.InputStream;
+import java.util.Locale;
+import java.util.Set;
+
+import org.eclipse.smarthome.ui.icon.IconSet.Format;
+
+/**
+ * An icon provider can provide {@link InputStream}s for icons.
+ * The source of the images can depend on the provider implementation.
+ * The byte stream represents a PNG or SVG image, depending on the format.
+ *
+ * The icon category corresponds to the list of available channel categories.
+ *
+ * In order to provide means to user interfaces to know, what kind of icon sets are available in the system (and offered
+ * by some icon provider), the provider can additionally provide a set of {@link IconSet}s.
+ *
+ * @author Kai Kreuzer - Initial contribution
+ */
+public interface IconProvider {
+
+    /**
+     * Returns a set of icon set definitions (meta-data) which this provider defines.
+     *
+     * @return a set of icon sets in the default locale
+     */
+    Set<IconSet> getIconSets();
+
+    /**
+     * Returns a set of localized icon set definitions (meta-data) which this provider defines.
+     *
+     * @param locale the locale to use for the results
+     * @return a set of icon sets in the requested locale
+     */
+    Set<IconSet> getIconSets(Locale locale);
+
+    /**
+     * determines whether this provider can deliver an icon for a given name
+     *
+     * @param category the icon category
+     * @param iconSetId the id of the icon set for which the icon is requested
+     * @param format the format of the stream (usually either png or svg)
+     * @return an Integer value defining the priority (higher is more important) or <code>null</code>, if this provider
+     *         cannot deliver an icon
+     */
+    Integer hasIcon(String category, String iconSetId, Format format);
```

Comment:
Let´s say I want to implement an IconProvider. What priority do I usually have to return? 0, 1? Does it has to be higher than some defined value? Like the priority of the default icons?